### PR TITLE
Rework the setup guides and Requirements

### PR DESCRIPTION
**Pull request by Christian-Rades**

### CODE REVIEW

File: guides/installation/README.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 
-## Production hosting
+## Setting up your own environment
 
-If you are looking for a guide to set up Shopware in a production environment, please follow the link below:
+Please refer to our [requirements](requirements.md) to install and configure the necessary services
+like a database and a webserver to a *nix system like Linux, macOS, wsl etc.
 
-{% page-ref page="flex.md" %}
+After that you can setup Shopware as a symfony flex project. [Guide](template.md)
```
Comment (shyim) [inline guides/installation/README.md:20]:
```suggestion
After that you can setup Shopware as a Symfony flex project. [Guide](template.md)
```

File: guides/installation/requirements.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+**SQL**
+
+* Tested on MySQL 5.7.21, and 8.0
+  * Only MySQL 8.0.20 in specific, is not compatible
+* Tested on MariaDB 10.3.22, 10.4 and 10.5
+  * MariaDB 10.3.29, 10.4.19, 10.5.10 are not compatible at the moment
+
+**JavaScript**
+
+* Node.js 12.21.0 or higher
```
Comment (shyim) [inline guides/installation/requirements.md:57]: We should directly update this to latest version

File: guides/installation/requirements.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  * ext-sodium
+  * ext-xml  
+  * ext-zip  
+  * ext-zlib
+* Composer 2.0 or higher
+
+**SQL**
+
+* Tested on MySQL 5.7.21, and 8.0
+  * Only MySQL 8.0.20 in specific, is not compatible
```
Comment (shyim) [inline guides/installation/requirements.md:51]: Maybe makes sense to create a warning block here

File: guides/installation/requirements.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
+# Requirements
+
+Before installing Shopware 6, you should take a quick look at the requirements to check if your local environment is capable of running it.
+
+## Operating System
+
+Shopware 6 is currently only supported on linux and macOS setups.
+Windows is only supported inside wsl.
```
Comment (shyim) [inline guides/installation/requirements.md:8]:
```suggestion
Windows is only supported inside WSL 2.
```

File: guides/installation/requirements.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+* Tested on MySQL 5.7.21, and 8.0
+  * Only MySQL 8.0.20 in specific, is not compatible
+* Tested on MariaDB 10.3.22, 10.4 and 10.5
+  * MariaDB 10.3.29, 10.4.19, 10.5.10 are not compatible at the moment
+
+**JavaScript**
+
+* Node.js 12.21.0 or higher
+* NPM 6.5.0 or higher
```
Comment (shyim) [inline guides/installation/requirements.md:58]: IDK what this is

File: guides/installation/requirements.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+## Webserver
```
Comment (shyim) [inline guides/installation/requirements.md:60]: I would like to make a Recommended Setup section

```
Webserver: Caddy
PHP: 8.1 (6.5  8.2)
Node: 16 (6.5 18)
Search: Opensearch 2.5.0
Queue: Rabbitmq
```

Comment (shyim) [reply guides/installation/requirements.md:60]: Maybe we can move the php ini settings from Performance Tweaks to here or link to them?

File: guides/installation/requirements.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+}
+```
+
+{% endtab %}
+
+
+{% tab title="Apache" %}
+
+{% hint style="info" %}
+`mod_headers` must be enabled to securely serve `.svg` files.
```
Comment (shyim) [inline guides/installation/requirements.md:157]: - mod_negotiation
- mod_rewrite
- mod_headers
- mod_deflate

File: guides/installation/requirements.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+Below you find the default configuration using either Caddy, Nginx or Apache as a websever.
+
+{% tabs %}
+{% tab title="Caddy" %}
+
+```text
+mydomain.com {
+  root * public
+  php_fastcgi 127.0.0.1:9000
+  file_server
```
Comment (shyim) [inline guides/installation/requirements.md:73]:
```
header {
        X-Frame-Options DENY
        Referrer-Policy no-referrer-when-downgrade
      }

      @svg {
        file
        path *.svg
      }

      header @svg Content-Security-Policy "script-src 'none'"

      @default {
        not path /theme/* /media/* /thumbnail/* /bundles/* /css/* /fonts/* /js/* /recovery/* /sitemap/*
      }

      root * public
      php_fastcgi @default unix/${config.languages.php.fpm.pools.web.socket}
      encode zstd gzip
      file_server
```

File: guides/installation/requirements.md [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+{% hint style="info" %}
+`mod_headers` must be enabled to securely serve `.svg` files.
+{% endhint %}
+
+```text
+<VirtualHost *:80>
+   ServerName "HOST_NAME"
+   DocumentRoot _SHOPWARE_LOCATION_/public
+   <Directory _SHOPWARE_LOCATION_>
+      Options Indexes FollowSymLinks MultiViews
```
Comment (shyim) [inline guides/installation/requirements.md:165]:
```suggestion
      Options -Indexes +FollowSymLinks +MultiViews
```

File: guides/installation/README.md [hunk 10]
```diff
@@ -1,16 +1,20 @@
...
 There are a couple of ways to get Shopware running on your system togehter with the most useful services for data storage and a webserver.
 
-* [devenv](devenv.md): Installation using devenv, built on top of Nix.
-* [Dockware](community/dockware.md): This is a managed docker setup for Shopware 6 by Shopware agency [dasistweb](https://www.dasistweb.de/).
+## Declaratively managed
+
+The installation with Devenv ([see the guide](devenv.md)) manages all necessary services. A description file in the source code manages the versions of these services.
+This setup works for linux and macOS.
+
+A community powered alternative for this is [Dockware](community/dockware.md).
```
Comment (fschmtt) [inline guides/installation/README.md:12]:
```suggestion
A community powered alternative to this is [Dockware](community/dockware.md).
```

File: guides/installation/README.md [hunk 11]
```diff
@@ -1,16 +1,20 @@
...
 This section discusses ways to set up Shopware on local machines so you can use it as a foundation for your development.
 
 There are a couple of ways to get Shopware running on your system togehter with the most useful services for data storage and a webserver.
 
-* [devenv](devenv.md): Installation using devenv, built on top of Nix.
-* [Dockware](community/dockware.md): This is a managed docker setup for Shopware 6 by Shopware agency [dasistweb](https://www.dasistweb.de/).
+## Declaratively managed
+
+The installation with Devenv ([see the guide](devenv.md)) manages all necessary services. A description file in the source code manages the versions of these services.
+This setup works for linux and macOS.
```
Comment (fschmtt) [inline guides/installation/README.md:10]:
```suggestion
This setup works for Linux (also WSL) and macOS.
```

File: guides/installation/README.md [hunk 12]
```diff
@@ -1,16 +1,20 @@
...
+
+A community powered alternative for this is [Dockware](community/dockware.md).
+This is a managed docker setup for Shopware 6 by Shopware agency [dasistweb](https://www.dasistweb.de/).
 
-## Production hosting
+## Setting up your own environment
 
-If you are looking for a guide to set up Shopware in a production environment, please follow the link below:
+Please refer to our [requirements](requirements.md) to install and configure the necessary services
+like a database and a webserver to a *nix system like Linux, macOS, wsl etc.
```
Comment (fschmtt) [inline guides/installation/README.md:18]:
```suggestion
like a database and a webserver to a *nix system like Linux, macOS, WSL etc.
```

File: guides/installation/requirements.md (hunk 5 above)
Comment (fschmtt) [reply guides/installation/requirements.md:58]: Depends on the Node version I guess
6.4 => Node 16, npm 6.x
6.5 => Node 18, npm 8.x

File: guides/installation/requirements.md (hunk 3 above)
Comment (fschmtt) [reply guides/installation/requirements.md:51]: Or a table with ✅ and 🚫 for easier visualization 🤔

Comment (Christian-Rades) [reply guides/installation/requirements.md]: I created a problematic version list.

File: guides/installation/requirements.md [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+
+    client_max_body_size 128M;
+
+    root __DOCUMENT_ROOT__/public;
+
+    # Shopware install / update
+    location /recovery/install {
+        index index.php;
+        try_files $uri /recovery/install/index.php$is_args$args;
+    }
```
[the remark this answers is not in the document]
Comment (Christian-Rades) [reply guides/installation/requirements.md:120]: This is currently for the 6.4, so it should still be necessary, no?

[the remark this answers is not in the document]
Comment (shyim) [reply guides/installation/requirements.md:120]: nope. we removed that installer even in 6.4

**Review by shyim — APPROVED** [review]

**Review by shyim — APPROVED** [review]

**Review by sushmangupta — CHANGES_REQUESTED** [review]

**Review by sushmangupta — APPROVED** [review]

**Comment by sushmangupta** [comment]
I see a small glitch. Please allow me to fix it before merging.

**Comment by sushmangupta** [comment]
> I see a small glitch. Please allow me to fix it before merging.

Done

**Pull request merged**